private/pkg/storage/storageos/bucket.go

```diff
@@ -404,6 +404,25 @@ func (w *writeObjectCloser) Close() error {
 	return err
 }
 
+// onceError is an object that will only store an error once.
+type onceError struct {
+	err atomic.Value
+}
+
+// Store stores the err.
+func (e *onceError) Store(err error) {
+	e.err.CompareAndSwap(nil, err)
+}
+
+// Load loads the stored error.
+func (e *onceError) Load() error {
+	err, ok := e.err.Load().(error)
+	if !ok {
+		return nil
+	}
+	return err
+}
```

Comment:
took this from a bufstream PR; could use a raw `atomic.Value` inline, but this prevents needing to do the type assertion at the callsite. 🤷.